### PR TITLE
Use `pytest.mark.parameterize` for simple validation tests

### LINKED ISSUE #7448: Refactoring test codes in `tests/utils/test_validation.py`
### Willingness to contribute

Yes. I can contribute this feature independently.

### Proposal Summary

Related #7447 

Using `pytest.mark.parametrize` for some test cases like `test_is_numeric` can make  it easier to find which case doesn`t work. Good for readability.

### Motivation

Code refactoring.
> #### What is the use case for this feature?

> #### Why is this use case valuable to support for MLflow users in general?

> #### Why is this use case valuable to support for your project(s) or organization?

> #### Why is it currently difficult to achieve this use case?

### Details

_No response_

### What component(s) does this bug affect?

- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/recipes`: Recipes, Recipe APIs, Recipe configs, Recipe Templates
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [ ] `area/server-infra`: MLflow Tracking server backend
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

### What interface(s) does this bug affect?

- [ ] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

### What language(s) does this bug affect?

- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

### What integration(s) does this bug affect?

- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

### PR DESCRIPTION
**Pull request by tsugumi-sys**
Signed-off-by: tsugumi-sys <tidemark0105@gmail.com>

## Related Issues/PRs

Resolve #7448 

## What changes are proposed in this pull request?

(Please fill in changes proposed in this fix)

- Use `pytest.mark.parameterize` for simple validation unit tests.
- NOT use in `test_validate_batch_log_limits` and `test_validate_batch_log_data` because readability gets rather worse.

### NOTE

If you use `pytest.mark.parameterize`, the pytest log generated for each cases like below.

image

## How is this patch tested?

- [x] Existing unit/integration tests
- [ ] New unit/integration tests
- [ ] Manual tests (describe details, including test results, below)

## Does this PR change the documentation?

- [x] No. You can skip the rest of this section.
- [ ] Yes. Make sure the changed pages / sections render correctly in the documentation preview.

### Is this a user-facing change?

- [x] No. You can skip the rest of this section.
- [ ] Yes. Give a description of this change to be included in the release notes for MLflow users.

(Details in 1-2 sentences. You can just refer to another PR with a description if this PR is part of a larger change.)

### What component(s), interfaces, languages, and integrations does this PR affect?

Components

- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/recipes`: Recipes, Recipe APIs, Recipe configs, Recipe Templates
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [ ] `area/server-infra`: MLflow Tracking server backend
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface

- [ ] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language

- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

Integrations

- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

<a name="release-note-category"></a>

### How should the PR be classified in the release notes? Choose one:

- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [x] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

@harupy

### CODE REVIEW

File: tests/utils/test_validation.py [hunk 1]
```diff
@@ -198,29 +238,64 @@ def test_validate_batch_log_data():
...
+        ("sqlite", True),
+        # error cases
+        ("MySQL", False),
+        ("mongo", False),
+        ("cassandra", False),
+        ("sql", False),
+        ("", False),
+    ],
+)
+def test_db_type(db_type, is_good_type):
```
Comment (harupy) [inline tests/utils/test_validation.py:295]: I'd consider splitting the test in this case.

Comment (tsugumi-sys) [reply tests/utils/test_validation.py:295]: How about like this?

```python
@pytest.mark.parametrize('db_type', ['mysql', 'mssql'])
def test_good_db_type(db_type):
  _validate_db_type_string(db_type)
```

Comment (harupy) [reply tests/utils/test_validation.py:295]: I'd name the function `test_validate_db_type_string_good`.

File: tests/utils/test_validation.py [hunk 2]
```diff
@@ -62,44 +62,61 @@ def test_path_not_unique(path, expected):
...
-
-
-def test_validate_run_id():
-    for good_id in [
+@pytest.mark.parametrize(
+    ("value", "expected"),
+    [
+        (0, True),
+        (0.0, True),
+        # error cases
```
Comment (harupy) [inline tests/utils/test_validation.py:70]:
```suggestion
        # Non-numeric cases
```

**Review by harupy — APPROVED** [review]

**Pull request merged**